cocotb_build_libs.py

```diff
@@ -38,6 +44,34 @@ def run(self):
 
         super().run()
 
+    def build_extensions(self):
+        for ext in self.extensions:
+            refactor = {"extra_link_args":[], "extra_compile_args":[]}
+
+            refactor["extra_compile_args"] += _extra_cxx_compile_args
+
+            rpaths = ext.library_dirs
+
+            ext.library_dirs.append(os.path.join(self.build_lib, "cocotb", "libs"))
+            if os.path.split(ext.name)[-1] == "simulator":
```

Comment:
Not a huge fan of having this knowledge here. Can we just add $ORIGIN to library_dirs?